test/load/traders/trader.js

```diff
@@ -45,6 +48,17 @@ class Trader {
     return this.client.connect();
   }
 
+  processTrades(trade) {
+    const index = this.trades[trade.tradeId];
+    if (index !== undefined) {
+      this.trades[index] = trade;
+    } else {
+      this.trades.unshift(trade);
+      this.trades[0].price = strategyConfig.price;
```

Comment:
This is the wrong way round - this line is changing the price of a trade, whereas the RandomTrader class needs to alter its marketPrice based on the price specified in the trade.

We need to keep the Trader base class clean of any strategy specific and derived class details. It would be better to define a set of 'virtual' functions in the Trader class, e.g.:
* onOrder() {}
* onTrade() {}
* onTransaction() {}

that can be overridden in the RandomTrader class to react to the given events.
The message handler (in initialise()) will then call these functions for each incoming item